### PR TITLE
Multi class branch

### PR DESCRIPTION
**Pull request by wangxinyu199306**
I have modified some files for my multi-class training. however, I might slightly changed your source code that may influence your original function.

1. I trained with full image , not the square image after crop. 
thus, I rewrite the "get_full_img_and_mask" function, so that it can return the same format as "get_imgs_and_masks".

2.add a new parameter "class_mark", in "train" function which is a list(e.g. if you have 5 classes in your task and the intensity on mask image is (32,64,96,128,160), the list should be [0,32,64,96,128,160]).
  
3.add function "split_mask_image" in utils.py, for split a n-class mask into n binary-masks

4.After Operate predict.py with "python predict.py -i image.png" , it will write 0.png, 1.png ,... n.png, depends how many classes in your task

### CODE REVIEW

**Comment by allenhamm** [comment]
How can I clone the multi class branch?

**Comment by dmitrysarov** [comment]
I suppose that if you consider multi class (not multi label) case, criterion should be changed to CrossEntropy as well as last activation from sigmoid to softmax.

**Comment by mamrouche** [comment]
@dmitrysarov I believe that you don't need a final softmax layer when using the CrossEntropyLoss() from Pytorch given that it is already built in it regarding the official documentation: [official doc](https://pytorch.org/docs/stable/nn.html#crossentropyloss).
You can directly pass the raw final feature map to the CrossEntropyLoss.

**Comment by milesial** [comment]
Closing this as it was implemented in the last commits.
